DCReplace/EventHandlers.cs

```diff
@@ -19,30 +23,66 @@ class EventHandlers
 
 		private Dictionary<Player, Vector3> PositionsToSpawn = new Dictionary<Player, Vector3>();
 
-		private List<Player> TryGet035()
+		private Player scp035PlayerReference = null;
+		private Exiled.API.Interfaces.IPlugin<Exiled.API.Interfaces.IConfig> scp035Plugin;
+		private Player scp966PlayerReference;
+		private Exiled.API.Interfaces.IPlugin<Exiled.API.Interfaces.IConfig> scp966Plugin;
+
+		private List<CoroutineHandle> currentReplacementCoroutines;
+
+
+
+		/// <summary>
+		/// Gets the last 035 player from Scp035 DLL logic, which is a hash of the player and their information.
+		/// </summary>
+		/// <param name="currPlayer"></param>
+		private void TryGet035(Player currPlayer)
 		{
-			List<Player> scp035 = null;
 
-			foreach (var plugin in Loader.Plugins)
+			Log.Info("Getting035");
+
+			if (scp035Plugin != null)
 			{
-				if (plugin.Name == "scp035")
-				{
-					try
-					{
-						scp035 = (List<Player>)Loader.Plugins.First(pl => pl.Name == "scp035").Assembly.GetType("scp035.API.Scp035Data").GetMethod("GetScp035s", BindingFlags.Public | BindingFlags.Static).Invoke(null, null);
-					}
-					catch (Exception e)
-					{
-						Log.Debug("Failed getting 035s: " + e);
-						scp035 = new List<Player>();
-					}
-				}
+
+				//Under assumption of only 1 035 allowed at one time. 
+				scp035PlayerReference = (Player)scp035Plugin?.Assembly?.GetType("scp035.API.Scp035Data")?.GetMethod("GetScp035IfRecentlyExisted", BindingFlags.Public | BindingFlags.Static)?.Invoke(null, new object[] { currPlayer });
+				return;
+			}
+
+
+
+			try
+			{
+
+				//Under assumption of only 1 035 allowed at one time. 
```

Comment:
The plugin is designed around the ability to have more than 1 035. In production it is configured to have 1

Reply:
Yes, I believe we discussed this after I had made these changes. I will modify it to be more generic

Reply:
Actually no, it pulls from a dictionary, it should already be generic